File: goldencheck/engine/sampler.py
"""Smart sampling for large datasets."""
from __future__ import annotations
import polars as pl

def maybe_sample(df: pl.DataFrame, max_rows: int = 100_000) -> pl.DataFrame:
    if len(df) <= max_rows:
        return df
    return df.sample(n=max_rows, seed=42)

File: packages/goldencheck-js/src/core/engine/sampler.ts
/**
 * Smart sampling for large datasets.
 * Port of goldencheck/engine/sampler.py.
 */

import type { TabularData } from "../data.js";

/** Return data unchanged if ≤ maxRows, otherwise deterministic sample. */
export function maybeSample(data: TabularData, maxRows: number = 100_000): TabularData {
  if (data.rowCount <= maxRows) return data;
  return data.sample(maxRows, 42);
}
